client/src/main/java/tds/session/Extern.java

```diff
@@ -0,0 +1,45 @@
+package tds.session;
+
+/**
+ * configuration properties for the session system
+ */
+public class Extern {
+    private String clientName;
+    private String environment;
+
+    public static class Builder {
```

Comment:
I'm not sure I see the benefit of the Builder pattern here.  I'm used to using it as a way of building a complex object and a lot of times it is around parsing some representation of the object back into the object itself.  It looks like the only use (at least right now) is for loading a POJO representation from the database.